src/librustc/lint/mod.rs

```diff
@@ -131,51 +131,104 @@ pub trait LintPass {
 // contains a few lint-specific methods with no equivalent in `Visitor`.
 pub trait LateLintPass: LintPass {
```

Comment:
I can almost do it (https://is.gd/TrZudu), I'm not sure why Rust can't figure out that a lifetime on a generic bound in the trait generics will outlive the lifetime of a method argument:

```rust
impl<'b, 'tcx: 'b, T: Foo<'b, 'tcx>> Helper<'tcx> for T {
    fn as_foo<'a>(&'a mut self) -> &'a mut Foo<'a, 'tcx> where 'tcx: 'a {
        self
    }
}
```

```
note: the lifetime 'a as defined on the block at 17:72...
note: ...does not necessarily outlive the lifetime 'b as defined on the block at 17:72
```